src/stores/ScannerStore.js

```diff
@@ -94,16 +152,23 @@ export default class ScannerStore extends Container<ScannerState> {
 
   async setTXRequest(txRequest, accountsStore) {
     this.setBusy();
-    console.log(txRequest);
-    if (!(txRequest.data && txRequest.data.rlp && txRequest.data.account)) {
+
+    const protocol = txRequest.data.data.crypto ? NetworkProtocols.SUBSTRATE : NetworkProtocols.ETHEREUM
+
+    if (protocol === NetworkProtocols.ETHEREUM && !(txRequest.data && txRequest.data.rlp && txRequest.data.account)) {
       throw new Error(`Scanned QR contains no valid transaction`);
     }
-    const tx = await transaction(txRequest.data.rlp);
-    const { ethereumChainId: networkKey = '1' } = tx;
 
+    if (protocol === NetworkProtocols.ETHERUEM) {
+      const tx = await transaction(txRequest.data.rlp);
+      const { ethereumChainId = 1 } = tx;
+      const networkKey = ethereumChainId;
+    }
+
+     // TODO cater for Substrate
     const sender = accountsStore.getById({
-      networkType: NetworkTypes.ETHEREUM,
-      networkKey,
+      protocol,
+      networkKey: networkKey || NetworkProtocols.SUBSTRATE,
```

Comment:
Not sure how we should do that.
for Ethereum `networkKey === ethereumChainId`, for Substrate we could set it to `networkKey === genesisHash`? Right now it's `s + networkPrefix`.. but that doesn't help.